presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -304,7 +304,7 @@ public PlanOptimizers(
                         ImmutableSet.<Rule<?>>builder()
                                 .addAll(new InlineSqlFunctions(metadata, sqlParser).rules())
                                 .addAll(new DesugarLambdaExpression().rules())
-                                .addAll(new SimplifyCardinalityMap().rules())
```

Comment:
Is it possible to reproduce the bug even without putting this optimizer ? Like what happens if we simply remove this rule ?

Reply:
Its not, the bug is in the optimizer

Reply:
the optimizer introduces the bug by rewriting the query incorrectly.  if we remove the rule it won't reproduce.